### PR TITLE
Bug 2051443: cnf-tests: verify minimal CPUs requirement for oslat test

### PR DESCRIPTION
**Pull request by Tal-or**
We should handle the amount of minimal CPUs for oslat test more carefully.
When HT disabled oslat needs 2 CPUs.

But when HT enabled oslat needs at least 4 CPUs:
One CPU should be used to run oslat's main thread.
The second is the sibling of the first one, which should be excluded from the list of the tested CPUs,
Because it might cause to false spikes (noisy-neighbor issue).
The third one is the actual CPU to be tested, but due to SMT alignment restriction we need its sibling too.
Four in total.

Signed-off-by: Talor Itzhak <titzhak@redhat.com>

### CODE REVIEW

File: cnf-tests/pod-utils/oslat-runner/main.go [hunk 1]
```diff
@@ -32,10 +32,24 @@ func main() {
...
 	mainThreadCPUs := selfCPUs.ToSlice()[0]
+	klog.Info("oslat main thread cpu: %d", mainThreadCPUs)
+
 	siblings, err := node.GetCPUSiblings(mainThreadCPUs)
 	if err != nil {
 		klog.Fatalf("failed to get main thread CPU siblings: %v", err)
 	}
+	klog.Info("oslat main thread's cpu siblings: %v", siblings)
+
+	// siblings > 0 means Hyper-threading enabled
```
Comment (shajmakh) [inline cnf-tests/pod-utils/oslat-runner/main.go:43]: can you elaborate why?

Comment (Tal-or) [reply cnf-tests/pod-utils/oslat-runner/main.go:43]: This is how Hyper-threading works, one physical core is treated as two logical.
two logical cores of the same physical one, are treated as siblings.
https://stackoverflow.com/questions/7274585/linux-find-out-hyper-threaded-core-id

**Review by fromanirh — APPROVED** [review]
/approve

**Comment by shajmakh** [comment]
@Tal-or looks good, thanks for the change. I expect there should be a following change on PAO/NTO side in case CPUs are less than 4:
https://github.com/openshift/cluster-node-tuning-operator/blob/master/test/e2e/performanceprofile/functests/4_latency/latency.go#L78

**Comment by Tal-or** [comment]
> @Tal-or looks good, thanks for the change. I expect there should be a following change on PAO/NTO side in case CPUs are less than 4: https://github.com/openshift/cluster-node-tuning-operator/blob/master/test/e2e/performanceprofile/functests/4_latency/latency.go#L78

Why? The check is system-dependent so it must be done inside the test pod.
The code you are referring to prior to the test pod creation.

**Comment by shajmakh** [comment]
> > @Tal-or looks good, thanks for the change. I expect there should be a following change on PAO/NTO side in case CPUs are less than 4: https://github.com/openshift/cluster-node-tuning-operator/blob/master/test/e2e/performanceprofile/functests/4_latency/latency.go#L78
> 
> Why? The check is system-dependent so it must be done inside the test pod which runs on the target system. The code you are referring to prior to the test pod creation.

because customers runs latency tools tests via the cnf-tests, and if they run them on HT env with less than 4 CPUs the test will fail to run properly because the test pod will not start running as it hits SMT error. Thus such cases should be skipped with a relevant message to the user.  Similarly to https://github.com/openshift/cluster-node-tuning-operator/blob/master/test/e2e/performanceprofile/functests/4_latency/latency.go#L96

**Comment by Tal-or** [comment]
> > > @Tal-or looks good, thanks for the change. I expect there should be a following change on PAO/NTO side in case CPUs are less than 4: https://github.com/openshift/cluster-node-tuning-operator/blob/master/test/e2e/performanceprofile/functests/4_latency/latency.go#L78
> > 
> > 
> > Why? The check is system-dependent so it must be done inside the test pod which runs on the target system. The code you are referring to prior to the test pod creation.
> 
> because customers runs latency tools tests via the cnf-tests, and if they run them on HT env with less than 4 CPUs the test will fail to run properly because the test pod will not start running as it hits SMT error. Thus such cases should be skipped with a relevant message to the user. Similarly to https://github.com/openshift/cluster-node-tuning-operator/blob/master/test/e2e/performanceprofile/functests/4_latency/latency.go#L96

This check runs in ```BeforeEach``` we'll hit that anyway, aren't we?

**Pull request merged**